packages/xstate-wallet/src/workflows/support-state.ts

```diff
@@ -16,57 +17,50 @@ Do we abort? Or do we try to reach consensus on a later state?
 */
 export const config: MachineConfig<HasChannelId, any, AnyEventObject> = {
   key: WORKFLOW,
-  initial: 'sendState',
+  initial: 'signState',
   states: {
-    sendState: {
+    signState: {
       entry: [
         assign<HasChannelId>({channelId: ({state}) => calculateChannelId(state)}),
         'spawnObserver'
       ],
-      invoke: {src: 'sendState'},
+      invoke: {src: 'signState'},
       on: {SUPPORTED: 'success'}
     },
     success: {type: 'final'}
   }
 };
 
-type Services = {sendState(ctx: HasChannelId): any};
+type Services = {signState(ctx: HasChannelId, event): any};
 
 type Options = {
   services: Services;
   actions: {spawnObserver: AssignAction<HasChannelId, any>};
 };
 
-const sendState = (store: Store) => async ({state, channelId}: HasChannelId) => {
+const signState = (store: Store) => async ({state, channelId}: HasChannelId) => {
   const entry = await store.getEntry(channelId);
-  const {isSupportedByMe, isSupported} = entry;
-  // TODO: Should these safety checks be performed in the store?
-  if (
-    // If we've haven't already signed a state, there's no harm in supporting one.
-    !isSupportedByMe ||
-    // If we've already supported this state, we might as well re-send it.
-    (isSupportedByMe && statesEqual(entry.latestSignedByMe, state)) ||
-    // Otherwise, we only send it if we haven't signed any new states.
-    (isSupported && entry.supported.turnNum.lt(state.turnNum)) ||
-    // We always support a final state if it matches the outcome that we have signed
-    (state.isFinal && outcomesEqual(state.outcome, entry.latestSignedByMe.outcome))
-  ) {
+  const {isSupportedByMe} = entry;
+  // We only sign the state if we haven't signed it already
+  if (!isSupportedByMe || !statesEqual(entry.latestSignedByMe, state)) {
     await store.signAndAddState(channelId, state);
-  } else {
-    throw 'Not safe to send';
+  } else if (statesEqual(entry.latestSignedByMe, state)) {
+    // The support state machine was started with a state that we already support
+    // That's fine but we output a warning in case that's unexpected
+    warn({state}, 'The state is already supported');
```

Comment:
1. I'm not 100% sure what to do here. Maybe you should send it again (what if it's supported in my store but not yours?)

2. What if it's not safe to send? We should probably be throwing, so that the parent machine catches (an explicit error) and (potentially) retries with a different state